tests/end_to_end/helpers.h

```diff
@@ -484,6 +484,135 @@ _xdp_context_destroy(
     free(context);
 }
 
+typedef class _test_global_helper
+{
+  public:
+    static uint64_t
+    _sample_get_pid_tgid()
+    {
+        return 9999;
+    }
+} test_global_helper_t;
+
+typedef class _test_sample_helper
+{
+  public:
+    static int64_t
+    _sample_ebpf_extension_helper_function1(_In_ const sample_program_context_t* context)
+    {
+        UNREFERENCED_PARAMETER(context);
+        return 0;
+    }
+
+    static int64_t
+    _sample_ebpf_extension_find(_In_ const void* buffer, uint32_t size, _In_ const void* find, uint32_t arg_size)
+    {
+        UNREFERENCED_PARAMETER(size);
+        UNREFERENCED_PARAMETER(arg_size);
+        return strstr((char*)buffer, (char*)find) - (char*)buffer;
+    }
+
+    static int64_t
+    _sample_ebpf_extension_replace(
+        _In_ const void* buffer, uint32_t size, int64_t position, _In_ const void* replace, uint32_t arg_size)
+    {
+        int64_t result = 0;
+        char* dest;
+        char* end = (char*)buffer + size - 1;
+        char* source = (char*)replace;
+        UNREFERENCED_PARAMETER(arg_size);
+
+        if (position < 0) {
+            result = -1;
+            goto Exit;
+        }
+
+        if (position >= size) {
+            result = -1;
+            goto Exit;
+        }
+
+        dest = (char*)buffer + position;
+        while (dest != end) {
+            if (*source == '\0') {
+                break;
+            }
+            *dest++ = *source++;
+        }
+
+    Exit:
+        return result;
+    }
+} test_sample_helper_t;
+
+// These are test sample context creation functions.
+static ebpf_result_t
+_sample_test_context_create(
+    _In_reads_bytes_opt_(data_size_in) const uint8_t* data_in,
+    _In_ size_t data_size_in,
+    _In_reads_bytes_opt_(context_size_in) const uint8_t* context_in,
+    _In_ size_t context_size_in,
+    _Outptr_ void** context)
+{
+    ebpf_result_t retval = EBPF_FAILED;
+    sample_program_context_t* sample_context = nullptr;
+    *context = nullptr;
+
+    // Data is not supported
```

Comment:
nit: inconsistent comment style.  Other comments in this file end in punctuation.  Here, 567, 603, don't.